### ISSUE TITLE
RSI does not appear to correctly trigger buy/sell (Solved)

### ISSUE THREAD

**Issue by mattlxdesign** [freqtrade/freqtrade#5595]
* Operating system: Ubuntu 20.04
  * Python Version: 3.8.10 
  * CCXT version: 1.55.28 
  * Freqtrade Version: 2021.8 

  * What happened?
I'm sure the bug is with me rather than freqtrade as I've only been using it for just over a week and I can't find any other issues mentioning this. In my quest to learn how the various indicators work I've found I cannot get buy/sell to trigger on RSI events as expected. Initially I hyperopted my buy/sell rsi values but then changed them to hard values while trying to work this out. I have no other indicators enabled.

Here are my populate_buy_trend and populate_sell_trend methods:

   ```
def populate_buy_trend(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
        dataframe.loc[
            (
                qtpylib.crossed_above(dataframe['rsi'], 30)
            ),
            'buy'] = 1
        return dataframe
```

    def populate_sell_trend(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
        dataframe.loc[
            (
                dataframe['rsi'] > 70
            ),
            'sell'] = 1
        return dataframe

### Steps to reproduce:

  1. Try any values for RSI to cross, >, < regardless of being hard values or variables
  2. run a Dry Run or Back Test
  3. Buy/Sell triggers do not occur as expected
  
### Observed Results:
When I run this strategy I get no sells and buys are when rsi is around 50 and then drops to around 40. Back testing would suggest sell triggers happen once in one or two days. This is using 15m candles. When doing a dry run I can find many candles where the rsi is over 70 with no sell being triggered and I wrote above about the observed buy triggers.

  * What did you expect to happen?
Buy to trigger when RSI crosses above 30.
Sell to trigger when RSI is above 70

Apologies if this should have been posted as a question rather than bug. I was in two minds about where to post it. Thanks to anyone who might be able to point out the glaringly obvious thing I'm missing....

**Comment by xmatthias** [comment]
With this strategy, buys will ONLY happen when RSI crosses above 30.
This means, it'll have to be <30 for at least one candle, and then cross the (imaginary) horizontal line at 30 from bottom to top.

Your example mentions values of 50 and 40 ... which are expected to do nothing with this strategy.

**Comment by mattlxdesign** [comment]
I'm sorry, I wasn't clear. What you describe - buys will ONLY happen when RSI crosses above 30. Is what I expected. What I observed - rsi is around 50 and then drops to around 40.
Which is absolutely not what I'm trying to achieve.

**Comment by xmatthias** [comment]
i'm not sure i understand the question, then.

If rsi is between 30 and 70, there'll be no action (apart from stoploss / roi).

**Comment by mattlxdesign** [comment]
The question is, for my learning process I want to trigger a buy event when rsi crosses above 30 and a sell event when rsi is above 70. I believe my above code should do that. What I'm seeing is buy events when the rsi cross down to 40 and sell events are so rare I can't find a common cause. And I don't understand how this is happening.

**Comment by xmatthias** [comment]
the most likely (and from experience, most frequent) reason is that you're having a strategy with the same name in the directory - so the strategy you think you're working on is a different one. (you can check this with `freqtrade list-strategies`).

An alternative is that you have the same method-names twice in the strategy (the latter one will win) - so your code is not used. While difficult to detect automatically - editors (like vscode with python plugin) will help you detect these things.

Other than that - your strategy seems to do what you expect.

**Comment by mattlxdesign** [comment]
Your `freqtrade list-strategies` suggestion is excellent! It's so useful! It didn't detect any double naming though. 
My editor of choice is PyCharm which does cry a bit when I'm writing code in a project directory it didn't create. Even without me letting it control package imports it's very good at pointing out my many errors. 
I'm totally stumped as to why rsi would not be working for me. It's got to be something really simple I'm missing.

**Comment by xmatthias** [comment]
best use `freqtrade plot-dataframe` (or freqUI) to plot the indicator - maybe it's calculated differently than how you think it is (just because it's called RSI doesn't mean it has to contain RSI ...).

**Comment by mattlxdesign** [comment]
I've been relying heavily on freqUI to visualise the candles along with indicators. It's brilliant! That with TradingView are excellent resources for understanding the indicators. 
freqUI is how I could read the rsi at 50ish dropping to 40ish being what was actually triggering a buy. As I say. I'm stumped

**Comment by xmatthias** [comment]
Depending on the way you use freqUI (via webserver, on a running bot) you can check 2 things
* is the timeframe correct (when running via webserver)
* is the plotted indicator correct
* is the strategy correct (mainly webserver, but also trade-mode might be running a different strategy than you think it is)

**Comment by mattlxdesign** [comment]
I think I've worked it out! And as expected it was entirely me. In my first attempt, with high confidence I'd decided to HyperOpt first and include a couple of trigger options, either crossing up or down on both buy and sell. For that I'd used an if statement and appended conditions to a list. Then a lamda function inside the dataframe to be returned -
```
dataframe.loc[
                reduce(lambda x, y: x & y, conditions),
                'sell'] = 1
```
When I was having trouble I removed the if statements and only kept one option for each but left them being appended to a list and the lamda as above. Turns out rsi doesn't works so well that way. Thanks so much for your help
